### ISSUE TITLE
Problems listing sub-directory

### ISSUE THREAD

**Issue by esell** [sagikazarmark/caddy-fs-s3#26]
I am having issues listing sub-directories (I realize that is not proper s3 lingo :) ) inside of my s3 bucket and was curious if you had seen this issue before?
My Caddyfile looks like this:
```
files.blah.com {
	route {
		file_server {
			browse /home/ec2-user/browse2.tpl
			fs s3 {
				bucket "mybucket"
				region "us-west-2"
				endpoint "s3.us-west-2.amazonaws.com"
				force_path_style
			}
		}
	}
}
```

My bucket structure looks like:
```
bucket
  |-dir1
    |-file1.txt
    |-file2.txt
  |-dir2
    |-file1.txt
    |-file2.txt
  |-dir3
    |-file1.txt
    |-file2.txt
```

When I access `files.blah.com` I get a listing of the bucket as expected. When I try to access a sub-directory though, for example `files.blah.com/dir1/`, I get a 404. What is interesting though, is if I put in the direct file path, say `files.blah.com/dir1/file1.txt`, the file loads just fine.

I turned the logging to DEBUG and it appears for some reason that the sub-directory is not being picked up as an actual directory, but instead a file?

Here is a log snippet for a successful request `files.blah.com`:
```
2023/02/25 16:42:01.238 DEBUG   http.handlers.file_server       sanitized path join     {"site_root": ".", "request_path": "/", "result": "."}
2023/02/25 16:42:01.308 DEBUG   http.handlers.file_server       no index file in directory      {"path": ".", "index_filenames": ["index.html", "index.txt"]}
2023/02/25 16:42:01.308 DEBUG   http.handlers.file_server       browse enabled; listing directory contents      {"path": ".", "root": "."}
```
Notice the `browse enabled; listing directory contents`? It looks like Caddy has check that if the path is a directory, it goes to listing mode.

Now here is a log snippet for an unsuccessful request of `files.blah.com/dir1/`:
```
023/02/25 16:42:46.294 DEBUG   http.handlers.file_server       sanitized path join     {"site_root": ".", "request_path": "/dir1/", "result": "dir1/"}
2023/02/25 16:42:46.358 DEBUG   http.log.error.log0     {id=w3n2ps2s0} fileserver.(*FileServer).notFound (staticfiles.go:601): HTTP 404.....(cut the rest of the error out as it is just HTTP stuff)
```

You'll see that the `browse enabled` is never triggered. I believe that for some reason Caddy is not detecting this as a directory and trying to treat it as a file. 

Any ideas on what might be going on here? I will poke around the code a bit, but I figured I'd share first in case I'm doing something incorrectly.

**Comment by esell** [comment]
Ok, so I figured it out! First off, the solution is to add:
`uri strip_suffix /`
Or at least that is what worked for me. 

I hooked up delve to debug what was going on, and from what I can tell, s3fs doesn't like when files have a `/` in them. What caddy (and in turn caddy-fs-s3) was doing, was passing the "directory" name as `dir1/` instead of `dir1`. s3fs has some code to validate the path (https://github.com/jszwec/s3fs/blob/main/fs.go#L131) that was failing with the `/` in there, so I just had caddy remove it from the uri and things started working.
